### DISCUSSION TITLE
Not able to publish a message to Kafka

### DISCUSSION THREAD

**Discussion by DrNykterstein10** [Netflix/conductor#3834] [Q&A] [answered]
I am new to Conductor as well as Kafka. I was trying to use the Kafka System task to publish a message to Kafka. I am using a prebuilt Docker image of Container to run it. My workflow execution keeps failing with the following error: `Task 1b2b9039-796d-11ee-bd7a-0242ac110002 failed with status: FAILED and reason: 'Failed to invoke kafka task due to: org.apache.kafka.common.errors.TimeoutException: Topic test not present in metadata after 500 ms.'`

I have verified the existence of the topic test using console. Could anyone help me figure out what the issue is?

I have the following workflow definition:

`{
  "ownerApp": null,
  "createTime": 1698920339862,
  "updateTime": 1698921859149,
  "createdBy": null,
  "updatedBy": null,
  "accessPolicy": {},
  "name": "KafkaWorkflow",
  "description": "Sample kafka workflow",
  "version": 5,
  "tasks": [
    {
      "name": "call_kafka",
      "taskReferenceName": "call_kafka",
      "description": null,
      "inputParameters": {
        "kafka_request": {
          "topic": "test",
          "value": "Publishing message to Kafka using conductor",
          "bootStrapServers": "localhost:9092"
        }
      },
      "type": "KAFKA_PUBLISH",
      "dynamicTaskNameParam": null,
      "caseValueParam": null,
      "caseExpression": null,
      "scriptExpression": null,
      "dynamicForkJoinTasksParam": null,
      "dynamicForkTasksParam": null,
      "dynamicForkTasksInputParamName": null,
      "startDelay": 0,
      "subWorkflowParam": null,
      "sink": null,
      "optional": false,
      "taskDefinition": {
        "ownerApp": null,
        "createTime": null,
        "updateTime": null,
        "createdBy": null,
        "updatedBy": null,
        "accessPolicy": {},
        "name": null,
        "description": null,
        "retryCount": 3,
        "timeoutSeconds": 0,
        "inputKeys": [],
        "outputKeys": [],
        "timeoutPolicy": "TIME_OUT_WF",
        "retryLogic": "FIXED",
        "retryDelaySeconds": 60,
        "responseTimeoutSeconds": 3600,
        "concurrentExecLimit": null,
        "inputTemplate": {},
        "rateLimitPerFrequency": 0,
        "rateLimitFrequencyInSeconds": 1,
        "isolationGroupId": null,
        "executionNameSpace": null,
        "ownerEmail": null,
        "pollTimeoutSeconds": null,
        "backoffScaleFactor": 1
      },
      "rateLimited": null,
      "asyncComplete": false,
      "loopCondition": null,
      "retryCount": null,
      "evaluatorType": null,
      "expression": null
    }
  ],
  "inputParameters": [],
  "outputParameters": {},
  "failureWorkflow": "",
  "schemaVersion": 2,
  "restartable": true,
  "workflowStatusListenerEnabled": false,
  "ownerEmail": "piyush1.rai@paytm.com",
  "timeoutPolicy": "ALERT_ONLY",
  "timeoutSeconds": 0,
  "variables": {},
  "inputTemplate": {}
}`

**Comment by Robban1980** [comment, marked as answer]
its likely `localhost` that is the issue, from a docker container to access localhost on the host you would call `host.docker.internal`
